sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/SymbolicExecution/TooManyInternalStatesException.cs

```diff
@@ -19,11 +19,9 @@
  */
 
 using System;
-using System.Runtime.Serialization;
 
 namespace SonarAnalyzer.Helpers.FlowAnalysis.Common
 {
-    [Serializable]
     public class TooManyInternalStatesException : Exception

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Update this implementation of 'ISerializable' to conform to the recommended serialization pattern. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS3925)